lib/webauth.js

```diff
@@ -277,6 +308,16 @@ module.exports = function(authManager, cookiePort, isSecurePort) {
         initZLUXSession(req);
         req.session.zlux.expirationTime = Date.now() + result.expms;
       }
+      for (let [authHandler, authObject] of Object.entries(result.categories)) {
+        if (authObject.success == false) {
+          for (let [authPlugin, authPluginObject] of Object.entries(authObject.plugins)) {
+            if (authPluginObject.reason == EXPIRED_PASSWORD) {
+              res.status(HTTP_STATUS_PRECONDITION_REQUIRED).json(result);
+              return;
+            }
+          }
+        }
+      }
       result.defaultCategory = authManager.defaultType;
```

Comment:
Put this above your for loop, because it should be in every reponse from now on.